### ISSUE TITLE
Screen Reader: Global: Search button announced without button role

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77338]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS, open the app
2. On the Inbox screen, swipe right to focus the Search button and note how it is announced

This is a global issue occurring in all screens with the Search button
Examples:
1. Profile > Status: the "Clear after" and "Vacation delegate" buttons
2. Profile > Contact method: the email address buttons
3. Profile > Timezone: the "Timezone" button
4. Profile > Address: the "Country" and "State" buttons
5. Preferences and subsections: "Priority mode", "Language", "Payment currency" and "Theme" buttons
6. Security: the "Two-factor authentication", "Merge accounts", "Report suspicious activity", "Close account" and "Add copilot" buttons
7. About: the "App download links", "View keyboard shortcuts" and "Report a bug" buttons
8. Wallet: the "Add bank account", added cards under "Assigned cards" and "enable wallet" buttons
9. Add bank > What are your account details?: the "Currency" button
10. Add bank > Check your info: all buttons with chevron
11. Expensify card: the "Report virtual card fraud" and "View transactions" buttons
12. Create Expense > Confirm Details: all buttons with chevron
13. Group chat > Details: all buttons with chevron
14. Create workspace > Confirm Workspace: the "Default currency" button
15. Create workspace > [USERNAME]'s Workspace: the "[USERNAME]'s expenses" button

## Expected Result:
VoiceOver should announce the button as a button, for example: "Search, button".
Users should understand that it is actionable and can be activated.

## Actual Result:
The Search button is missing its accessible role so VoiceOver only announces it as "Search" with no indication that it is an interactable element.

## Area issue was found in:
Inbox

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Update the programmatic role of the Search button to ensure it is announced as such using accessibilityRole="button".
The button should be announced as "Search, button".

Refer to:
https://appt.org/en/docs/react-native/samples/accessibility-role
https://reactnative.dev/docs/accessibility#accessibilityrole

## Environment(s)
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)

## Screenshots/Videos
<img width="1179" height="2431" alt="Image" src="https://github.com/user-attachments/assets/f2bfe66f-521e-442a-b515-b9264ea66bb6" />

https://github.com/user-attachments/assets/e15663a5-9c87-45dc-bfe4-df63d7c36cc7

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005812
**Issue reported by:** NA
**Slack conversation:** NA
